src/game/behaviors/spawn_star.inc.c

```diff
@@ -61,32 +65,36 @@ void bhv_star_spawn_init(void) {
 
 void bhv_star_spawn_loop(void) {
     switch (o->oAction) {
-        case 0:
+        case SPAWN_STAR_ARC_CUTSCENE_ACT_START:
             o->oFaceAngleYaw += 0x1000;
-            if (o->oTimer > 20)
-                o->oAction = 1;
+            if (o->oTimer > 20) {
+                o->oAction = SPAWN_STAR_ARC_CUTSCENE_ACT_GO_TO_HOME;
+            }
             break;
 
-        case 1:
+        case SPAWN_STAR_ARC_CUTSCENE_ACT_GO_TO_HOME:
             obj_move_xyz_using_fvel_and_yaw(o);
             o->oStarSpawnVelY += o->oVelY;
             o->oPosY = o->oStarSpawnVelY + sins((o->oTimer * 0x8000) / 30) * 400.0f;
             o->oFaceAngleYaw += 0x1000;
             spawn_object(o, MODEL_NONE, bhvSparkleSpawn);
             cur_obj_play_sound_1(SOUND_ENV_STAR);
             if (o->oTimer == 30) {
-                o->oAction = 2;
+                o->oAction = SPAWN_STAR_ARC_CUTSCENE_ACT_BOUNCE;
                 o->oForwardVel = 0;
+                // Set to exact home coordinates
+                o->oPosX = o->oHomeX;
+                o->oPosZ = o->oHomeZ;
```

Comment:
what does this do?

Reply:
It snaps the star position to the exact coordinates at the end of its arc. For example, in vanilla, the THI Wiggler star isn't always centered on the floor, despite its destination (home) position being centered.